src/modules/nationalId/index.ts

```diff
@@ -2,30 +2,41 @@
  * Check National-id validation
  * @category National id
  * @method verifyIranianNationalId
- * @param  {String?}          nationalId [String of national id - like this: 1111111111]
+ * @param  {String?}          nationalId [String of national id - like this: 0018465986]
  * @return {Boolean}                    [valid or no]
  */
-function verifyIranianNationalId(nationalId?: string | number): boolean | null | undefined {
-	if (!nationalId) return;
-	else {
-		let code = nationalId.toString();
 
-		if (!code.match(/^\d{10}$/)) return false;
-		code = ("0000" + code).substr(code.length + 4 - 10);
+function verifyIranianNationalId(nationalId?: string | number): boolean | undefined {
+	if (!nationalId) return;
+	let code = nationalId.toString();
+	const codeLength = code.length;
+	const monoDigits = [
```

Comment:
I thought about this array and I realized if someone tries to run this function by an onChange event, iteration this array will be a performance issue(o(n) time complexity) which is not a good idea. Now my idea is to use an object instead of an array with o(1) time complexity.
Sample:
```javascript
const notAllowedDigits = {
'222222222': true, 
... 
} 

if(code in notAllowedDigits) return false
```

Agree?

Reply:
Yes, I agree with you. It's done.